src/main/java/org/openstreetmap/atlas/generator/AtlasGeneratorHelper.java

```diff
@@ -106,115 +95,9 @@ public static ConfiguredTaggableFilter getTaggableFilterFrom(
-                    {
-                        final Optional<FileNotFoundException> fileNotFound = ExceptionSearch
-                                .find(FileNotFoundException.class).within(e);
-                        if (fileNotFound.isPresent())
-                        {
-                            // It's possible there is no Atlas file for a given shard
-                            logger.debug("No Atlas file found at {}", path);
-                            return Optional.empty();
-                        }
-                        else
-                        {
-                            // There is something else going on, re-throw and continue
-                            throw e;
-                        }
-                    }
-
-                    // Before making the move, check again if file is there or not
-                    if (!fileFromTemporaryDirectory.exists())
-                    {
-                        try
-                        {
-                            Files.move(Paths.get(temporaryLocalFile.getPath()),
-                                    Paths.get(fileFromTemporaryDirectory.getPath()),
-                                    StandardCopyOption.ATOMIC_MOVE);
-                        }
-                        catch (final FileAlreadyExistsException e)
-                        {
-                            logger.warn("Failed to rename file, but file exists already.", e);
-                        }
-                        catch (final Exception e)
-                        {
-                            logger.warn("Failed to rename file on local disk.", e);
-                        }
-                    }
-                }
-                catch (final Exception e)
-                {
-                    logger.warn("Failed to cache file on local disk.", e);
-                }
-            }
-
-            // If we were able to find the file on local disk, then load from there
-            if (fileFromTemporaryDirectory.exists())
-            {
-                logger.debug("Atlas exists - Cache Hit: {}",
-                        fileFromTemporaryDirectory.getAbsolutePath());
-                return loadAtlas(fileFromTemporaryDirectory);
-            }
-            else
-            {
-                logger.debug("Falling back to Atlas file hosted on {} for shard {}.",
-                        atlasDirectory, shard.getName());
-                final String path = SparkFileHelper.combine(atlasDirectory,
-                        String.format("%s%s", getAtlasName(country, shard), ATLAS_EXTENSION));
-                final Resource fileFromNetwork = FileSystemHelper.resource(path, sparkContext);
-                try
-                {
-                    return loadAtlas(fileFromNetwork);
-                }
-                catch (final Exception e)
-                {
-                    final Optional<FileNotFoundException> fileNotFound = ExceptionSearch
```

Comment:
Do we no longer hit this case or is it handled by the cache itself?

Reply:
The cache handles this. See my comment below.